artichoke-backend/src/extn/core/string/string.rb

```diff
@@ -574,25 +574,10 @@ def encoding
   end
 
   # https://ruby-doc.org/core-3.0.2/String.html#method-i-end_with-3F
-  def end_with?(*suffixes)
-    suffixes.each do |suffix|
-      case suffix
-      when String
-        return true if suffix.empty?
-        return true if self[-suffix.length..-1] == suffix
-      when Regexp
```

Comment:
Ooo thank you. Can you add a test to the string functional tests to assert that `String#end_with?` raises `ArgumentError` when given a `Regexp`? You can run this whole file with MRI by invoking it `ruby path/to/string_test.rb` to ensure we have parity.

Maybe one case with a single regexp, one case with a regexp in the set of args where none match, and one case with a regexp in the set of args where an arg earlier in the sequence matches?

https://github.com/artichoke/artichoke/blob/dac4c8c9280a00350ae2421628b6f1b1710171b7/artichoke-backend/src/extn/core/string/string_test.rb#L17-L23

Reply:
I've added the test you've suggested to that file. It's made me doubt putting the `start_with?` regexp test in rust though. Should that new test have been in the `string_test.rb` too, or is it fine where it is?

Reply:
I think it's fine where it is. It allows us to gate the test on the core-regexp feature.